metamorph/plugins/morph_resultsdb.py

```diff
@@ -0,0 +1,225 @@
+
+    def query_resultsdb(self, url, url_options=dict, attempt=3):
+        """
+        This method queries resultsDB with given url and url_option variable
+
+        :param url -- resultdb api url
+        :param url_options -- dictionary of wanted options
+        :param attempt -- number of query tries if some problem occurs
+
+        :returns -- Queried data
+        """
+        logging.debug('Running resultsbd API query with this url: {0} and options {1}'.format(url, url_options))
+        response = requests.get(url, params=url_options)
+        if 300 <= response.status_code <= 400:
+            if attempt > 0:
+                time.sleep(self.MINUTE + self.MINUTE)  # Sleeping for two minutes
+                return self.query_resultsdb(url, url_options, attempt - 1)
+            else:
+                raise ResultsDBApiException("ERROR: Unable to access resultsdb site after 3 attempts")
+        elif response.status_code >= 400:
+            logging.error("ERROR: Unable to access resultsdb site.")
+            raise ResultsDBApiException("ERROR: Unable to access resultsdb site.")
+        return response.json()
+
+    def get_resultsdb_data(self, job_name="", limit=10):
+        """
+        Method for getting data from resultsDB
+
+        :param self.url_options -- class dictionary of url options
+        :param job_name -- job name which will be searched in resultsDB
+        :param limit -- Limit for amount of queried pages from resultsDB
+        :returns -- List of queried data
+        """
+        i = 0
+        next_page = ""
+        queried_data = []
+        if job_name:
+            self.url_options['job_names'] = job_name
+        while next_page is not None and self.TIMEOUT_LIMIT and limit > i:
+            self.url_options['page'] = i
+            response_data = self.query_resultsdb(self.resultsdb_api_url, self.url_options)
+            if not response_data['data']:
+                logging.info("job name has not published results to resultsDB yet, sleeping...")
+                time.sleep(self.MINUTE)  # Sleeping for 1 minute
+                self.TIMEOUT_LIMIT -= self.MINUTE
+            else:
+                i += 1
+                next_page = response_data['next']
+                queried_data += response_data['data']
+        if self.TIMEOUT_LIMIT == 0:
+            raise ResultsDBApiException("Timeout limit reached and no data were queried.")
+        return queried_data
+
+    @staticmethod
+    def setup_output_data(resultsdb_data):
+        """
+        Method for setting up data from get_job_by_nvr_and_tier method.
+        Data needs to be in dictionary where keys will be job names
+
+        :param resultsdb_data -- data from get_job_by_nvr_and_tier
+        :returns -- dictionary where keys are job names and their values are list of queried data
+        """
+        formatted_output = {}
+        for single_job in resultsdb_data:
+            formatted_output[single_job['job_names']] += single_job
```

Comment:
resultsdb_data is list of queried resultsdb data (dictionaries). Therefore it can't be that way.